lib/bamboo/adapters/sendgrid_adapter.ex

```diff
@@ -89,6 +56,34 @@ defmodule Bamboo.SendgridAdapter do
     end
   end
 
+  defp raise_api_error(message), do: raise(ApiError, message: message)
+  defp raise_api_error(params, response) do
+    filtered_params = params |> Plug.Conn.Query.decode |> Map.put("key", "[FILTERED]")
+
+    message = """
+    There was a problem sending the email through the SendGrid API.
```

Comment:
@drapergeek Makes a lot more sense than the current implementation, I will improve this and let you know when it is done.